modules/geochemistry/src/userobjects/NodalVoidVolume.C

```diff
@@ -0,0 +1,226 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NodalVoidVolume.h"
+#include "Assembly.h"
+#include "libmesh/parallel_sync.h"
+
+registerMooseObject("GeochemistryApp", NodalVoidVolume);
+
+InputParameters
+NodalVoidVolume::validParams()
+{
+  InputParameters params = ElementUserObject::validParams();
+  params.addCoupledVar("porosity", 1.0, "Porosity");
+  params.addCoupledVar(
+      "concentration",
+      "A concentration variable.  This is only used to determine the finite-element type of your "
+      "concentration variable.  The default is linear Lagrange.  Therefore, if you are using "
+      "linear-lagrange variables you do not need to supply this input");
+  // All elements that share a node with the elements owned by this processor are known about by
+  // this UserObject
+  params.addRelationshipManager("ElementPointNeighborLayers",
+                                Moose::RelationshipManagerType::GEOMETRIC |
+                                    Moose::RelationshipManagerType::ALGEBRAIC |
+                                    Moose::RelationshipManagerType::COUPLING,
+                                [](const InputParameters &, InputParameters & rm_params) {
+                                  rm_params.set<unsigned short>("layers") = 1;
+                                });
+  params.addClassDescription(
+      "UserObject to compute the nodal void volume.  Take care if you block-restrict this "
+      "UserObject, since the volumes of the nodes on the block's boundary will not include any "
+      "contributions from outside the block.");
+  return params;
+}
+
+NodalVoidVolume::NodalVoidVolume(const InputParameters & parameters)
+  : ElementUserObject(parameters),
+    _porosity(coupledValue("porosity")),
+    _rebuilding_needed(true),
+    _nodal_void_volume(),
+    _my_pid(processor_id()),
+    _nodes_to_receive(),
+    _nodes_to_send(),
```

Comment:
```suggestion
    _my_pid(processor_id()),
```
One last thing. You're initializing these with their default constructors already, so you can just remove them.

Reply:
OK